release_notes/v2_vs_v3.adoc

```diff
@@ -0,0 +1,107 @@
+
+Flexible linking means you can link any two arbitrary components together. As
+long as one component can export environment variables and the second component
+can consume values from those environment variables, and with potential variable
+name transformation, you can link together any two components without having to
+change the images they are based on. So, the best containerized implementation
+of your desired database and web framework can be consumed directly rather than
+you having to fork them both and rework them to be compatible.
+
+The means you can build anything on OpenShift. And that is OpenShift's
+primary aim: to be a container-based platform that lets you build entire
+applications in a repeatable life cycle.
+
+== Cartridges vs Images
+
+In OpenShift v3, an link:../architecture/core_concepts/containers_and_images.html#docker-images[image] has replaced OpenShift v2's concept of a cartridge.
+
+Cartridges in OpenShift v2 were the focal point for building applications. Each cartridge provided the required libraries, source code, build mechanisms, connection logic, and routing logic along with a preconfigured environment to run the components of your applications.
+
+However, cartridges came with disadvantages. With cartridges, there was no clear distinction between the developer content and the cartridge content, and you did not have ownership of the home directory on each gear of your application. Also, cartridges were not the best distribution mechanism for large binaries. While you could use external dependencies from within cartridges, doing so would lose the benefits of encapsulation.
+
+From a packaging perspective, an image performs more tasks than a cartridge, and provides better encapsulation and flexibility. However, cartridges also included logic for building, deploying, and routing, which do not exist in images. In OpenShift v3, these additional needs are met by link:../architecture/core_concepts/builds_and_image_streams.html#source-build[Source-to-Image (S2I)] and link:../architecture/core_concepts/templates.html[template configuration].
+
+*Dependencies*
+
+In OpenShift v2, cartridge dependencies were defined with `*Configure-Order*` or `*Requires*` in
+a cartridge manifest. OpenShift v3 uses a declarative model where
+link:../architecture/core_concepts/pods_and_services.html#pods[pods] bring
+themselves in line with a predefined state. Explicit dependencies that are
+applied are done at runtime rather than just install time ordering.
+
+For example, you might require another service to be available before you start.
+Such a dependency check is always applicable and not just when you create the
+two components. Thus, pushing the dependency into a runtime dependency enables
+the system to stay healthy over time.
```

Comment:
I don't understand the last sentence.  Suggest "Thus, pushing dependency resolution to runtime enables..." or maybe s/pushing/confining/, and  s/check/resolution/ in the previous sentence.  (The concept of "resolution" includes "check" plus actions to fulfill the required relationship (i.e., bring up a service) should the check fail.)